mmselfsup/models/algorithms/barlowtwins.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional
+
+import torch
+import torch.nn as nn
+
+from ..builder import ALGORITHMS, build_backbone, build_head, build_neck
+from .base import BaseModel
+
+
+@ALGORITHMS.register_module()
+class BarlowTwins(BaseModel):
+    """BarlowTwins.
+
+    Implementation of `Barlow Twins: Self-Supervised Learning via Redundancy
+    Reduction <https://arxiv.org/abs/2103.03230>`_.
+    Part of the code is borrowed from:
+    `<https://github.com/facebookresearch/barlowtwins/blob/main/main.py>`_.
+
+    Args:
+        backbone (dict): Config dict for module of backbone.
+        neck (dict): Config dict for module of deep features to compact
+            feature vectors. Defaults to None.
+        head (dict): Config dict for module of loss functions.
+            Defaults to None.
+        init_cfg (dict): Config dict for weight initialization.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 backbone: object,
+                 neck: object = None,
+                 head: object = None,
+                 init_cfg: Optional[dict] = None,
+                 **kwargs) -> None:
+        super(BarlowTwins, self).__init__(init_cfg)
+        assert neck is not None
+        self.encoder = nn.Sequential(
```

Comment:
This is no need to create an extra ``encoder``, just using ``backbone`` and ``neck`` is OK